scrapy_autoextract/providers.py

```diff
@@ -1,114 +1,229 @@
         )
+        crawler.engine.close_spider(crawler.spider, "account_disabled")
 
-        try:
-            response = await request_raw(
-                [request],
-                api_key=api_key,
-                endpoint=endpoint,
-                max_query_error_retries=max_query_error_retries
-            )
-        except Exception:
-            self.stats.inc_value(f"autoextract/{page_type}/error/request")
-            raise
 
-        data = response[0]
+def get_concurrent_requests_per_domain(settings: Settings):
+    """Return the configured concurrent request per domain from settings"""
+    concurrency = settings.getint("CONCURRENT_REQUESTS_PER_DOMAIN", 8)
+    if concurrency < 1:
+        raise ValueError("Invalid 'CONCURRENT_REQUESTS_PER_DOMAIN' "
+                         f"value: {concurrency}")
+    return concurrency
 
-        if "error" in data:
-            self.stats.inc_value(f"autoextract/{page_type}/error/query")
-            raise QueryError(data["query"], data["error"])
 
-        self.stats.inc_value(f"autoextract/{page_type}/success")
-        return self.provided_class(data=data)
+class AutoExtractProvider(PageObjectInputProvider):
+    """Provider for AutoExtract data"""
+    page_type_class: ClassVar[Type]
 
-    @classmethod
-    def register(cls):
-        """Register this provider for its provided class on scrapy-poet
-        registry. This will make it possible to declare provided class as
-        a callback dependency when writing Scrapy spiders.
-        """
-        register(cls, cls.provided_class)
+    # pageType requested when only html is required
+    page_type_class_for_html: ClassVar[AutoExtractData] = AutoExtractProductData
+    html_query_attribute: ClassVar[str] = "fullHtml"
 
     @classmethod
-    def get_page_type(cls) -> str:
-        """Page type is defined by the class attribute `item_key` available on
-        `autoextract_poet.page_inputs` classes.
-        """
-        return cls.provided_class.item_key
+    def provided_classes(cls, type_: Callable) -> bool:
+        return (inspect.isclass(type_)
+                and issubclass(type_, (AutoExtractData, AutoExtractHtml)))
 
+    def __init__(self, crawler: Crawler):
+        """Initialize provider storing its dependencies as attributes."""
+        self.crawler = crawler
+        self.settings = crawler.spider.settings
+        self.task_manager = get_autoextract_task_manager(crawler)
+        self.aiohttp_session = None
+        self.crawler.signals.connect(self.close_aiohttp_session,
+                                     signal=signals.spider_closed)
+        retries_count = self.settings.getint(
+                "AUTOEXTRACT_MAX_QUERY_ERROR_RETRIES", 3)
+        self.common_request_kwargs = dict(
+            api_key=self.settings.get("AUTOEXTRACT_USER"),
+            endpoint=self.settings.get("AUTOEXTRACT_URL"),
+            max_query_error_retries=retries_count,
+            session=self.aiohttp_session
```

Comment:
I don't think it works as intended: session is set to None here, and when self.aiohttp_session is created later, self.common_request_kwargs still has session=None. It seems `common_request_kwargs` may be better implemented as a property.